src/Juvix/Backends/LLVM/Codegen/Graph.hs

```diff
@@ -61,28 +61,33 @@ isBothPrimary ∷
     HasState "varTab" VariantToType m
   ) ⇒
   m Operand.Operand
-isBothPrimary = Block.defineFunction Type.i1 "is_both_primary" args $
+isBothPrimary = Block.defineFunction Types.bothPrimary "is_both_primary" args $
   do
     -- TODO ∷ should this call be abstracted somewhere?!
     -- Why should Ι allocate for every port?!
-    mainPort ← allocaNumPortsStatic False (Operand.ConstantOperand (C.Int 32 0))
+    mainPort ← mainPort
     -- TODO ∷ Make sure findEdge is in the environment
     edge ← Block.externf "find_edge"
     nodePtr ← Block.externf "node_ptr"
     node ← load nodeType nodePtr
     port ← call portType edge (Block.emptyArgs [node, mainPort])
```

Comment:
Are we reading into `mainPort` here (pointer receiver)?

It seems like we're allocating an (empty?) port then using it as a parameter, but maybe I'm misreading.

Reply:
mainPort is just an alias for the allocaNumPortStatic call.

So it's just a constantOperand, that allocates the pointer if it needs to be a pointer (it doesn't here)